src/overlays/actors/ovl_En_Bat/z_en_bat.c

```diff
@@ -33,99 +40,521 @@ const ActorInit En_Bat_InitVars = {
+        this->yawTarget = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
+    } else if (finishedRotStep && (Rand_ZeroOne() < 0.015f)) {
+        this->yawTarget = this->actor.shape.rot.y +
+                          (((s32)((f32)0x1000 * Rand_ZeroOne()) + 0x1000) * ((Rand_ZeroOne() < 0.5f) ? -1 : 1));
+    }
+
+    finishedRotStep = Math_ScaledStepToS(&this->actor.shape.rot.x, this->pitchTarget, 0x100);
+
+    if ((this->actor.bgCheckFlags & 1) || (this->actor.depthInWater > -40.0f)) {
+        this->pitchTarget = -0x1000;
+    } else if (this->actor.world.pos.y < (this->actor.home.pos.y - 100.0f)) {
+        this->pitchTarget = -((s32)((f32)0x800 * Rand_ZeroOne()) + 0x800);
+    } else if ((this->actor.home.pos.y + 100.0f) < this->actor.world.pos.y) {
+        this->pitchTarget = (s32)((f32)0x800 * Rand_ZeroOne()) + 0x800;
+    } else if ((finishedRotStep) && (Rand_ZeroOne() < 0.015f)) {
+        this->pitchTarget += (s16)(((s32)((f32)0x400 * Rand_ZeroOne()) + 0x400) * ((Rand_ZeroOne() < 0.5f) ? -1 : 1));
+        this->pitchTarget = CLAMP(this->pitchTarget, -0x1000, 0x1000);
+    }
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if ((this->actor.xzDistToPlayer < 300.0f) && (this->timer == 0) && (Player_GetMask(play) != PLAYER_MASK_STONE) &&
+        (sNumberAttacking < BAD_BAT_MAX_NUMBER_ATTACKING) &&
+        (!(this->paramFlags & BAD_BAT_PARAMFLAG_CHECK_HEIGHTREL) || (fabsf(this->actor.playerHeightRel) < 150.0f))) {
+        EnBat_SetupDiveAttack(this);
+    }
+}
+
+void EnBat_SetupDiveAttack(EnBat* this) {
+    this->collider.base.atFlags |= AT_ON;
+    this->timer = 300;
+    this->actor.speedXZ = 4.0f;
+    sNumberAttacking++;
+    this->actionFunc = EnBat_DiveAttack;
+}
+
+void EnBat_DiveAttack(EnBat* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s32 isFacingPlayer;
+    Vec3f preyPos;
+
+    EnBat_StepAnimation(this, 2);
+    isFacingPlayer = Actor_IsFacingPlayer(&this->actor, 0x2800);
+
+    if (isFacingPlayer) {
+        s16 pitchTarget;
+
+        preyPos.x = player->actor.world.pos.x;
+        preyPos.y = player->actor.world.pos.y + 20.0f;
+        preyPos.z = player->actor.world.pos.z;
+
+        pitchTarget = Actor_PitchToPoint(&this->actor, &preyPos);
+        pitchTarget = CLAMP(pitchTarget, -0x3000, 0x3000);
+        Math_SmoothStepToS(&this->actor.shape.rot.x, pitchTarget, 2, 0x400, 0x40);
+    } else {
+        Math_SmoothStepToS(&this->actor.shape.rot.x, -0x800, 2, 0x100, 0x10);
+    }
+    if (isFacingPlayer || (this->actor.xzDistToPlayer > 80.0f)) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 4, 0xC00, 0xC0);
+    }
+
+    this->timer--;
+
+    if ((this->timer == 0) || (this->collider.base.atFlags & AT_HIT) || (Player_GetMask(play) == PLAYER_MASK_STONE) ||
+        (this->actor.bgCheckFlags & 1) || (player->stateFlags1 & 0x800000) || (this->actor.depthInWater > -40.0f)) {
```

Comment:
```suggestion
        (this->actor.bgCheckFlags & 1) || (player->stateFlags1 & PLAYER_STATE1_800000) || (this->actor.depthInWater > -40.0f)) {
```
You could update the one in Firefly too, but I'm *hoping* Anghelo's PR gets that.